symengine/real_mpfr.cpp

```diff
@@ -1033,6 +1033,22 @@ class EvaluateMPFR : public Evaluate
         mpfr_exp(t.get_mpfr_t(), x_, MPFR_RNDN);
         return real_mpfr(std::move(t));
     }
+    virtual RCP<const Basic> floor(const Basic &x) const override
+    {
+        SYMENGINE_ASSERT(is_a<RealMPFR>(x))
+        mpfr_srcptr x_ = down_cast<const RealMPFR &>(x).i.get_mpfr_t();
+        mpfr_class t(mpfr_get_prec(x_));
+        mpfr_rint_floor(t.get_mpfr_t(), x_, MPFR_RNDD);
```

Comment:
It should be, I wasn't sure of the double rounding requirement. It is clear to me now.